llama-index-integrations/llms/llama-index-llms-alibabacloud-aisearch/llama_index/llms/alibabacloud_aisearch/base.py

```diff
@@ -0,0 +1,173 @@
+        while True:
+            try:
+                return func(*args, **kwargs)
+            except TeaException as e:
+                if e.code == "Throttling.RateQuota":
+                    time.sleep(wait_seconds)
+                else:
+                    raise
+
+    return wrap
+
+
+def aretry_decorator(func, wait_seconds: int = 1):
+    async def wrap(*args, **kwargs):
+        while True:
+            try:
+                return await func(*args, **kwargs)
+            except TeaException as e:
+                if e.code == "Throttling.RateQuota":
+                    await asyncio.sleep(wait_seconds)
+                else:
+                    raise
+
+    return wrap
+
+
+class AlibabaCloudAISearchLLM(CustomLLM):
+    """
+    For further details, please visit `https://help.aliyun.com/zh/open-search/search-platform/developer-reference/text-generation-api-details`
+    """
+
+    _client: Client = PrivateAttr()
+    _options: RuntimeOptions = PrivateAttr()
+
+    aisearch_api_key: str = Field(default=None, exclude=True)
+    endpoint: str = None
+
+    service_id: str = "ops-qwen-turbo"
+    workspace_name: str = "default"
+
+    temperature: float = 0.5
+    top_k: float = 1
+    additional_kwargs: Dict[str, Any] = Field(default_factory=dict)
+
+    read_timeout: int = 60000
+    connection_timeout: int = 5000
+
+    def __init__(
+        self, endpoint: str = None, aisearch_api_key: str = None, **kwargs: Any
+    ) -> None:
+        def messages_to_prompt(messages: Sequence[ChatMessage]):
+            results = []
+            for message in messages:
+                message = GetTextGenerationRequestMessages(
+                    content=message.content, role=message.role
+                )
+                results.append(message)
+            return results
+
+        super().__init__(messages_to_prompt=messages_to_prompt, **kwargs)
+        self.aisearch_api_key = get_from_param_or_env(
+            "aisearch_api_key", aisearch_api_key, "AISEARCH_API_KEY"
+        )
+        self.endpoint = get_from_param_or_env("endpoint", endpoint, "AISEARCH_ENDPOINT")
+
+        config = AISearchConfig(
+            bearer_token=self.aisearch_api_key,
+            endpoint=self.endpoint,
+            protocol="http",
+        )
+
+        self._client = Client(config=config)
+
+        self._options = RuntimeOptions(
+            read_timeout=self.read_timeout, connect_timeout=self.connection_timeout
+        )
+
+    @property
+    def metadata(self) -> LLMMetadata:
+        """Get LLM metadata."""
+        return LLMMetadata(model_name=self.service_id, is_chat_model=True)
+
+    @property
+    def _default_params(self) -> Dict[str, Any]:
+        return {
+            "temperature": self.temperature,
+            "top_k": self.top_k,
+            **self.additional_kwargs,
+        }
+
+    @llm_completion_callback()
+    @retry_decorator
+    def complete(self, messages: Any, **kwargs: Any) -> CompletionResponse:
```

Comment:
complete is meant to take a single string, not a list of messages

Did you mean to implement the `chat`/`stream_chat` methods instead?

Reply:
Thank you for your guidance. I have made the necessary adjustments and implemented the `chat`/`stream_chat` methods as intended.